### DISCUSSION TITLE
sync_root_with_cwd appears to be broken

### DISCUSSION THREAD

**Discussion by SOberhoff** [nvim-tree/nvim-tree.lua#2690] [Q&A] [answered]
### Description

If one opens a new buffer after having changed the current working directory through nvim-tree's `sync_root_with_cwd` option, then `:pwd` while in the new buffer will print the directory neovim was originally opened in, not the new current working directory.

### Neovim version

```text
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1703358377
```

### Operating system and version

macOS 14.3.1

### nvim-tree version

d52fdeb

### Clean room replication

```lua
require("nvim-tree").setup({sync_root_with_cwd = true})
```

### Steps to reproduce

1. Launch neovim in directory `foo`.
2. Use <C+]> to navigate to `foo/bar`.
3. Open `foo/bar/README.md`
4. Run `:pwd`

### Expected behavior

`:pwd` returns `foo/bar`.

### Actual behavior

`:pwd` returns `foo`.

**Comment by alex-courtis** [comment]
`sync_root_with_cwd` is used to change the tree's root on `:cd` or similar:

```help
*nvim-tree.sync_root_with_cwd*
Changes the tree root directory on `DirChanged` and refreshes the tree.
  Type: `boolean`, Default: `false`
```

In your case, a `:cd bar` would do the job.

`:pwd` in the tree window and the file's window will show `.../bar`

However, I'm not quite sure that's what you're after...

**Comment by alex-courtis** [comment, marked as answer]
`actions.change_dir.global` will set the global cwd when changing directories in the tree:

```help
*nvim-tree.actions.change_dir*
vim |current-directory| behaviour.

    *nvim-tree.actions.change_dir.enable*
    Change the working directory when changing directories in the tree.
      Type: `boolean`, Default: `true`

    *nvim-tree.actions.change_dir.global*
    Use `:cd` instead of `:lcd` when changing directories.
    Consider that this might cause issues with the
    |nvim-tree.sync_root_with_cwd| option.
      Type: `boolean`, Default: `false`

    *nvim-tree.actions.change_dir.restrict_above_cwd*
    Restrict changing to a directory above the global cwd.
      Type: `boolean`, Default: `false`
```

**Comment by alex-courtis** [comment]
Better documentation of all these changing directory behaviours is needed: #2405
